ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -80,15 +84,15 @@ public AlterTableCompactOperation(DDLOperationContext context, AlterTableCompact
       Optional<String> partitionName =  partitionMap.keySet().stream().findFirst();
       partitionName.ifPresent(compactionRequest::setPartitionname);
     }
-    List<CompactionResponse> compactionResponses =
+    Map<CompactionResponse, String> compactionResponses =
         initiatorBase.initiateCompactionForTable(compactionRequest, table.getTTable(), partitionMap);
-    for (CompactionResponse compactionResponse : compactionResponses) {
+    for (Map.Entry<CompactionResponse, String> compactionResponseEntry : compactionResponses.entrySet()) {
+      CompactionResponse compactionResponse = compactionResponseEntry.getKey();
       if (!compactionResponse.isAccepted()) {
-        String message;
         if (compactionResponse.isSetErrormessage()) {
-          message = compactionResponse.getErrormessage();
           throw new HiveException(ErrorMsg.COMPACTION_REFUSED, table.getDbName(), table.getTableName(),
-              "CompactionId: " + compactionResponse.getId(), message);
+              compactionResponseEntry.getValue() == null ? "" :
+                  "(partition=" + compactionResponseEntry.getValue() + ")", compactionResponse.getErrormessage());
```

Comment:
Do we have a test case that verifies that partition is present in the Exception? If not can you please add one.

Reply:
Added new test case